src/gnssro/gnssro_AWSopendataNetcdf2ioda.py

```diff
@@ -0,0 +1,495 @@
+    obs_data[('impactParameterRO', "MetaData")] = np.float32(assign_values(np.asarray(impact, dtype=ioda_float_type)))
+    obs_data[('sensorAzimuthAngle', "MetaData")] = np.float32(assign_values(np.asarray(azim, dtype=ioda_float_type)))
+    obs_data[('bendingAngle', "ObsValue")] = np.float32(assign_values(np.asarray(bang, dtype=ioda_float_type)))
+
+    local_nlocs = len(lats)
+    #   switch ascending/descending values following IODA defination
+    if setting == 1:
+        obs_data[('satelliteAscendingFlag', "MetaData")] = np.array(np.repeat(0, local_nlocs), dtype=ioda_int_type)
+    elif setting == 0:
+        obs_data[('satelliteAscendingFlag', "MetaData")] = np.array(np.repeat(1, local_nlocs), dtype=ioda_int_type)
+    else:
+        obs_data[('satelliteAscendingFlag', "MetaData")] = np.array(np.repeat(int_missing_value, local_nlocs), dtype=ioda_int_type)
+
+    meta_data_types = def_meta_types()
+
+    for k, v in profile_meta_data.items():
+        if isinstance(v, np.int64):
+            obs_data[(k, 'MetaData')] = np.array(np.repeat(v, local_nlocs), dtype=np.int64)
+        elif isinstance(v, int):
+            obs_data[(k, 'MetaData')] = np.array(np.repeat(v, local_nlocs), dtype=ioda_int_type)
+        elif v.dtype.kind == 'f':
+            obs_data[(k, 'MetaData')] = np.array(np.repeat(v, local_nlocs), dtype=ioda_float32_type)
+        else:
+            print('do not know type')
+            sys.exit()
+
+    # set record number (multi file procesing will change this)
+    if record_number is None:
+        nrec = 1
+    else:
+        nrec = record_number
+    obs_data[('sequenceNumber', 'MetaData')] = np.array(np.repeat(nrec, local_nlocs), dtype=ioda_int_type)
+
+    if add_qc:
+        good = quality_control(profile_meta_data, np.array(impact_height, float), np.array(lats, float), np.array(lons, float))
+        if len(good) == 0:
+            return{}
+            # exit if entire profile is missing
+        for k in obs_data.keys():
+            obs_data[k] = obs_data[k][good]
+    return obs_data
+
+
+def get_meta_opendata(f):
+
+    # get some of the global attributes that we are interested in
+    meta_data_keys = def_meta_opendata()
+
+    # these are the MetaData we are interested in
+    profile_meta_data = {}
+    for k, v in meta_data_keys.items():
+        profile_meta_data[k] = np.array(f[v], dtype=ioda_float_type)
+
+    year = f.attrs['year'][0]
+    month = f.attrs['month'][0]
+    day = f.attrs['day'][0]
+    hour = f.attrs['hour'][0]
+    minute = f.attrs['minute'][0]
+    second = f.attrs['second'][0]
+
+    dtg = ("%4i-%.2i-%.2iT%.2i:%.2i:%.2iZ" % (year, month, day, hour, minute, second))
+    this_datetime = datetime.strptime(dtg, "%Y-%m-%dT%H:%M:%SZ")
+    time_offset = round((this_datetime - epoch).total_seconds())
+    profile_meta_data['dateTime'] = np.int64(time_offset)
+    WMO_sat_ID = get_WMO_satellite_ID(f.filename)
```

Comment:
here can pass the `f.attrs['leo']`

```suggestion
    WMO_sat_ID = get_WMO_satellite_ID(f.attrs['leo'])
```

Reply:
done!